src/vs/workbench/contrib/notebook/browser/services/notebookEditorServiceImpl.ts

```diff
@@ -137,6 +161,7 @@ export class NotebookEditorWidgetService implements INotebookEditorService {
 			// NEW widget
 			const instantiationService = accessor.get(IInstantiationService);
 			const widget = instantiationService.createInstance(NotebookEditorWidget, creationOptions ?? getDefaultNotebookCreationOptions(), initialDimension);
+			this._onDidAddNotebookEditorWidget.fire(widget);
```

Comment:
`NotebookEditorWidget` is calling `addNotebookEditor` directly so I guess we can directly listen to `onNotebookEditorAdd` event.